index.bs

```diff
@@ -1678,32 +1678,34 @@ partial interface MLGraphBuilder {
     </div>
 </div>
 
-### resample ### {#api-mlgraphbuilder-resample}
-Resample the tensor values from the source to the destination dimensions according to the scaling factors.
+### resample2d ### {#api-mlgraphbuilder-resample2d}
+Resample the tensor values from the source to the destination spatial dimensions according to the scaling factors.
 <script type=idl>
 enum MLInterpolationMode {
   "nearest-neighbor",
   "linear"
 };
 
-dictionary MLResampleOptions {
+dictionary MLResample2dOptions {
   MLInterpolationMode mode = "nearest-neighbor";
   sequence<float> scales;
   sequence<long> sizes;
+  MLInputOperandLayout layout = "nchw";
 };
 
 partial interface MLGraphBuilder {
-  MLOperand resample(MLOperand input, optional MLResampleOptions options = {});
+  MLOperand resample2d(MLOperand input, optional MLResample2dOptions options = {});
 };
 </script>
-<div algorithm=resample>
+<div algorithm=resample2d>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input 4-D tensor.
-        - *options*: an optional {{MLResampleOptions}}. The optional parameters of the operation.
+        - *options*: an optional {{MLResample2dOptions}}. The optional parameters of the operation.
             - *mode*: an {{MLInterpolationMode}}. The interpolation algorithm used to fill the output tensor values.
                 If not set, it is assumed to be the *Nearest Neighbor* interpolation.
-            - *scales*: a sequence of {{float}} of length 4. Each value represents the scaling factor used to scale in each input dimensions.
-            - *sizes*: a sequence of {{long}} of length 4. The target sizes for each input dimensions. When the target sizes are specified, the *options.scales* argument is ignored as the scaling factor values are derived from the target sizes of each input dimension.
+            - *scales*: a sequence of {{float}} of length 2. Each value represents the scaling factor used to scale in each spatial dimensions of input, [scale_height, scale_width]. If not set, the values are assumed to be [1.0, 1.0].
```

Comment:
Alternative to "If not set, the values are assumed to be [1.0, 1.0]." is to use a [dictionary member default value](https://heycam.github.io/webidl/#dfn-dictionary-member-default-value) for scales in IDL:

`sequence<float> scales = [1.0, 1.0];`

What would be the expected behavior when only either scale_height or scale_width is set?

Reply:
> `sequence<float> scales = [1.0, 1.0];`

Looks good, I'll address it in the new commit.

> What would be the expected behavior when only either scale_height or scale_width is set?

Since the spec says, it should be a sequence of {{float}} of length 2, it would throw error when only either scale_height or scale_width is set.

Reply:
> sequence<float> scales = [1.0, 1.0];

Unfortunately, It triggers a bikeshed error of the bot: https://github.com/webmachinelearning/webnn/pull/205/checks?check_run_id=3495820911

```
 [INFO] Build bikeshed document "index.bs" (gh)…
    $ bikeshed --die-on=link-error spec "index.bs" "index.bs.built.html" 
      FATAL ERROR: IDL SYNTAX ERROR LINE: 9 - expected ";" skipped: " = [1.0, 1.0]"
       ✘  Did not generate, due to fatal errors
    Failed
  Error: Process completed with exit code 2.
```

Reply:
[only an empty array `[]`](https://heycam.github.io/webidl/#index-prod-DefaultValue) is a valid default for a sequence in an IDL fragment; I think the prose version is the best we can do

Reply:
Thanks @dontcallmedom, @huningxin let's revert to the prose definition then.

Reply:
Thanks @dontcallmedom , fixed in the new commit.

Reply:
I looped in @dontcallmedom in https://github.com/webmachinelearning/webnn/pull/204#discussion_r702866958 that I believe is superseded by this PR, but the same questions still apply.

I'll unblock my review, since we can make the prose updates separately.